src/views/home/HueLightsList.vue

```diff
@@ -0,0 +1,314 @@
+<template>
+  <br />
+  <div
+    id="Lights"
+    v-for="[id, light] in hue_lights_map"
+    style="width: 33%; margin-top: 1%"
+    v-bind:key="light"
+  >
+    <CCard>
+      <CCardBody>
+        <CCardTitle>{{ light.label }}</CCardTitle>
+        <CFormSwitch
+          size="xl"
+          v-bind:id="light.label"
+          :checked="light.on == true"
+          v-on:click="light.switchState()"
+          track-by="$index"
+          v-bind:key="light.on"
+          style="width: 14%"
+        />
+        <br />
+        <div v-if="light.is_color == true">
+          <CFormInput
+            type="color"
+            id="exampleColorInput"
+            value="#563d7c"
+            title="Choose your color"
+          />
+          <br />
+        </div>
+        <div v-if="light.group">
+          <CFormSelect
+            aria-label="Select"
+            v-on:click="light.updateSelectedLight()"
+            v-model="light.value"
+            :options="light.group"
+          >
+          </CFormSelect>
+          <br />
+        </div>
+
+        <input
+          :disabled="light.on == false"
+          track-by="$index"
+          v-bind:key="light"
+          type="range"
+          id="vol"
+          name="vol"
+          min="0"
+          max="254"
+          v-model="light.bright"
+          style="width: 100%"
+          v-on:input="light.changeBright(id)"
+        />
+      </CCardBody>
+    </CCard>
+  </div>
+</template>
+
+<script>
+/* eslint-disable */
+let   lock_state    = false;
+const group_regex   = /(^[a-zA-Z ]*)([ \d]*\d$)/
+const api_url       = process.env.VUE_APP_HUE_BRIDGE_URL; 
+const hue_app_token = process.env.VUE_APP_HUE_TOKEN;
+const hue_is_color  = new Map([
+  ["LWA004", false],
+  ["LTG002", false],
+  ["LCT015", true],
+  ["LWA009", false],
+]);
+
+class HueLight {
+  constructor(name, on = false, id, bright, model_id) {
+    this.label    = name;
+    this.on       = on;
+    this.value    = id;
+    this.model_id = model_id;
+    this.is_color = hue_is_color.get(this.model_id);
+    this.bright   = bright;
+  }
+
+  switchState(force = false) {
+    if (lock_state === false || force === true) {
+      lock_state = true;
+      this.on = !this.on;
+      fetch("http://" + api_url + "/api/" + hue_app_token + "/lights/" + this.value + "/state", {
+        method: 'PUT',
+        body: JSON.stringify({
+          on: this.on,
+        })
+      })
+        .then((response) => response.json())
+        .then((data) => {
+          lock_state = false;
+        })
+    }
+  }
+
+  changeBright(force = false) {
+    if (lock_state === false || force === true) {
+      lock_state = true;
```

Comment:
Je comprends mtn à quoi sert lock_state, t'as l'air de l'utiliser pour faire en sorte que tant que y'a une requête en cours, l'utilisateur ne peut pas faire de nouvelles modifications. Je comprends la démarche mais c'est pas optimal sur plusieurs points:
- Tu intégre la logique de ton UI dans la classe qui définit une lampe hue
- T'as une variable qui risque de se retrouver dans un état bloqué car si ton appel API échoue, `lock_state` ne sera jamais remis à true

Là ce qu'il aurait fallu faire c'est gérer l'asynchrone avec async/await. L'idée c'est que tu ne vas plus avoir besoin de te faire chier avec les then et catch et l'aspect asynchrone de ton appel. Tu vas pouvoir appeler ton fetch comme si c'etait une fonction synchrone, ça veut dire que t'aura plus besoin de ton trick la avec lock_state.
Je sais pas trop comment ça marche avec vuejs mais j'ai trouvé un article qui peut t'aider https://medium.com/swlh/how-to-use-async-and-await-with-vue-js-apps-33132aa0838b


UPDATE: je viens de voir que ta variable `lock_state` elle est vraiment utilisée dans tous les appels API, c'est vraiment hyper dangereux psk tu vas avoir plusieurs fonctions qui vont opérer sur la meme variable. Il faut absolument que tu migre ça vers du async/await